src/map/npc.cpp

```diff
@@ -5507,16 +5528,21 @@ int npc_parsesrcfile(const char* filepath)
 			break;
 		}
 
+		bool has_script = false;
+
+		if (count > 3 && strncasecmp(w2, "script", 6) == 0)
+			has_script = true;
+
 		if( strcmp(w1,"-") !=0 && strcasecmp(w1,"function") != 0 )
 		{// w1 = <map name>,<x>,<y>,<facing>
 			char mapname[MAP_NAME_LENGTH_EXT];
 			int count2;
 
 			count2 = sscanf(w1,"%15[^,],%6hd,%6hd[^,]",mapname,&x,&y);
-			
+
 			if ( count2 < 1 ) {
 				ShowError("npc_parsesrcfile: Invalid script definition in file '%s', line '%d'. Skipping line...\n * w1=%s\n * w2=%s\n * w3=%s\n * w4=%s\n", filepath, strline(buffer,p-buffer), w1, w2, w3, w4);
-				if (strcasecmp(w2,"script") == 0 && count > 3) {
+				if (has_script) {
 					if ((p = npc_skip_script(p,buffer,filepath)) == NULL)
 						break;
 				}
```

Comment:
While you're at it, could you clean up this checks a bit?, get rid of all those curly braces and change the nulls to nullptr, so it'd be a bit easier to read :P